Modix.Services/ErrorHandling/AuthResult.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Humanizer;
+using Modix.Common.ErrorHandling;
+using Modix.Data.Models.Core;
+
+namespace Modix.Services.ErrorHandling
+{
+    /// <summary>
+    /// A <see cref="ServiceResult"/> representing a failure to execute a request due to the user missing claims
+    /// </summary>
+    public class AuthResult : ServiceResult
+    {
+        private readonly IReadOnlyDictionary<AuthorizationClaim, bool> _hasClaimLookup;
+
+        /// <summary>
+        /// The claims that were required for the request
+        /// </summary>
+        public IReadOnlyList<AuthorizationClaim> RequiredClaims { get; private set; }
+        /// <summary>
+        /// The claims that the current user has
+        /// </summary>
+        public IReadOnlyList<AuthorizationClaim> CurrentClaims { get; private set; }
+
+        /// <summary>
+        /// Creates an <see cref="InsufficientRankResult"/> using the given collections of <see cref="AuthorizationClaim"/>
```

Comment:
Should be `AuthResult` instead of `InsufficientRankResult`

Reply:
Fixed